ntc_templates/templates/cisco_xr_show_bgp_vrf_all_neighbors_advertised-routes.textfsm

```diff
@@ -0,0 +1,24 @@
+Value NETWORK (\d+\.\d+\.\d+\.\d+)
+Value PREFIX_LENGTH (\d+)
+Value NEXT_HOP (\d+\.\d+\.\d+\.\d+)
+Value FROM ((\d+\.\d+\.\d+\.\d+)|\S+)
+Value AS_PATH (.+)
+Value Filldown ROUTE_DISTINGUISHER (\d+\:\d+)
+Value Filldown VRF (\S+)
+
+Start
+  ^\S{3}\s+\S{3}\s+\d{1,2}\s+\d+:\d+:\d+
+  ^VRF:\s+\S+:\d+$$
+  ^-+$$
+  ^%\s+Neighbor\s+not\s+found$$
+  ^Network\s+Next\s+Hop\s+From\s+AS\s+Path$$
+  ^Route\s+Distinguisher\:\s+${ROUTE_DISTINGUISHER}\s+\(default\s+for\s+vrf\s+${VRF}\)\s*$$ -> ROUTES
+  ^Processed\s+
+  ^. -> Error
+
+ROUTES
+  ^${NETWORK}\/${PREFIX_LENGTH}\s+${NEXT_HOP}\s+${FROM}\s+${AS_PATH} -> Record
+  ^\s*$$ -> Start
+  ^. -> Error
```

Comment:
Several items here:
* Remove unneeded backslash before a colon
* Anchor the routes line with double dollar signs
* Avoid empty dictionary/match when hitting the `Processed` summary lines

Based on my testing, this works as expected. :grin:
If this is helpful, click the Commit Suggestion button to accept my edit and provide commit credit. Thank you!

```suggestion
  ^Route\s+Distinguisher:\s+${ROUTE_DISTINGUISHER}\s+\(default\s+for\s+vrf\s+${VRF}\)\s*$$ -> ROUTES
  ^. -> Error

ROUTES
  ^${NETWORK}\/${PREFIX_LENGTH}\s+${NEXT_HOP}\s+${FROM}\s+${AS_PATH}$$ -> Record
  ^\s*$$
  ^Processed\s+ -> Continue.Clearall
  ^Processed\s+ -> Start
  ^. -> Error
```